app/controllers/products_controller.rb

```diff
@@ -0,0 +1,29 @@
+class ProductsController < ApplicationController
+  before_action :set_product, only: [:show, :add_to_cart]
+
+  def add_to_cart
+    session[:order_id] ||= Order.create.id
+
+    # this should perhaps be a method in the order model
+    unless Order.find(session[:order_id]).already_has_product?(@product.id)
+      OrderItem.create(product_id: @product.id, order_id: session[:order_id], quantity_ordered: 1)
+    end
+
+    redirect_to cart_path # this should redirect to product show page
+  end
+
+  def index
+    @products = Product.all
+  end
+
+  def show
+    @reviews = Product.find(params[:id]).reviews
```

Comment:
The `before_action` is creating `@product` for you, so you can simplify this by doing `@reviews = @product.reviews`. It'd save you a query and clean up this code a bit.